.github/workflows/checkbox-snap-daily-builds.yml

```diff
@@ -0,0 +1,62 @@
+name: checkbox snap daily build
+
+on:
+  schedule:
+    - cron: '00 04 * * *'
+  workflow_dispatch:
+
+jobs:
+  snap:
+    strategy:
+      matrix:
+        type: [classic, uc]
+        releases: [16, 18, 20, 22]
+    runs-on: ubuntu-latest
+    env:
+      SERIES: series_${{ matrix.type }}${{ matrix.releases }}
+      SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.SNAPCRAFT7_CREDS }}
+    name: Checkbox snap for series ${{ matrix.type }}${{ matrix.releases }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check for checkbox snap new commits
+        run: |
+          git rev-list --abbrev-commit --pretty=oneline HEAD --not $(git rev-list -n1 --before="24 hours" --first-parent HEAD) -- checkbox-snap
+          [[ ! -z $(git rev-list --abbrev-commit --pretty=oneline HEAD --not $(git rev-list -n1 --before="24 hours" --first-parent HEAD) -- checkbox-snap) ]]
+      - name: Copy over the common files for series ${{ matrix.type }}${{ matrix.releases }}
+        run: |
+          cd checkbox-snap/
+          ./prepare_${{ matrix.type }}.sh $SERIES
+      - name: add LP credentials
+        run: |
+          mkdir -p ~/.local/share/snapcraft/provider/launchpad/
+          echo '${{ secrets.LP_CREDS }}' > ~/.local/share/snapcraft/provider/launchpad/credentials
+          git config --global user.email "robot@lists.canonical.com"
+          git config --global user.name "Certification bot"
+      - uses: snapcore/action-build@v1
+        id: snapcraft
+        with:
+          path: checkbox-snap/series_${{ matrix.type }}${{ matrix.releases }}
+          snapcraft-channel: 7.x/stable
+          snapcraft-args: remote-build --build-on amd64,arm64,armhf,i386 --launchpad-accept-public-upload
```

Comment:
Just checking for my own interest... this build that is performed by `snapcraft remote-build ...` is it strictly "private"? I assume not, but not problem in this case?

Reply:
It's not, all builds triggered via `snapcraft remote-build` are public. On success the ephemeral snap recipe on lp is cleanup up automatically but during the build or if it failed, it persists. See https://launchpad.net/~ce-certification-qa/+snaps

Reply:
Ref: https://snapcraft.io/docs/remote-build

> Prerequisites
> Prospective snaps need to be open source, as the code will be publicly available, and you’ll need a [Launchpad account](https://login.launchpad.net/+new_account).